splunk/Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM docker.io/splunk/splunk:latest AS builder
+
+ENV SPLUNK_START_ARGS=--accept-license
+
+EXPOSE 8080
+EXPOSE 8089
+
+USER root
+
+COPY tests/test_logs /opt/splunk/test_logs
+COPY tests/indexes.conf /opt/splunk/etc/apps/search/local/indexes.conf
+COPY scripts/index_verification.sh /opt/splunk/index_verification.sh
+COPY scripts/pre-run.sh /opt/splunk/pre-run.sh
+
+RUN microdnf install -y jq && \
+    microdnf clean all
+
+# USER ansible    # root user is still needed for removing files
+RUN bash /opt/splunk/pre-run.sh
+
+FROM builder AS splunk
```

Comment:
the whole point of doing multistage here is to start the final image from a clean base so that the sins you did in the build stage no longer hunt you. Doing "FROM builder" beats this purpose....

Reply:
Not sure I understand you then. which image should the second one start from, splunk as well? we decided on performing such "sins" because we need to assure we have the logs indexed before running any test on the container...

Reply:
It should state from the same clean image that you started the build phase from.
And yes you should copy the files, with the usually copy commands between stages.

The whole point of using multistage here is to do this:
1. Start from a clean image
2. Mess it up with some process (E.g. indexing)
3. Remove all the mess by restarting from clean image again
4. Copy just the files we need from the messed up image so we end up with a clean image

Reply:
Frankly if you didn't understand this, why were you trying to use multi-stage in the first place? just because I said so?

Reply:
I did understand what you meant when you suggested it, but did not fully understand the implementation itself. 
--
as for copying the files - I'm not sure which of the 1500+ files changed are the one relevant for us, 
but just to try, I copied the whole Splunk directory. funny enough, it works (I get the splunk instance running), but even with everything copied, the indexing seem to revert..

Reply:
This means only the some_log.log file, which is the one I'm copying to the container in the first place. I'm pretty sure this file is left unchanged (I checked), so hopefully we are not reversing the indexing... (for now it seems like either we are, or the indexing never happend)

Reply:
See `indexes.conf`, you list 3 files there:
```
coldPath = $SPLUNK_DB/$_index_name/colddb 
 homePath = $SPLUNK_DB/$_index_name/db 
 thawedPath = $SPLUNK_DB/$_index_name/thaweddb
```
Seems likely to me those are the files the index is stored in...

Reply:
I thought so too, but $SPLUNK_HOME/var/lib/splunk (which is the default for $SPLUNK_DB) does not exist, and $SPLUNK_DB leads to nowhere... I couldn't find the `$_index_name` directory at all.

Reply:
try to see what happens wen you run the image and create the index normally. These variables may be set by the startup script and not be preset in the image

Reply:
Done, it is finally working.
Building & Running the container now should get us a Splunk container with some logs attached to test_index index.